### ISSUE TITLE
core: expose programmatic api

### ISSUE THREAD

**Issue by arekkas** [ory-am/editor#267]
At the moment, there is no way to programmatically change the layout mode, for example. We should decide if the editor should really use the redux `<Provider>` or if this is the task of the developer integrating with us.

There are various things floating in my mind here:

* We could remove the redux Provider-dependency, but
  * Setting redux up is not straight forward and will not make sense for applications other than react
  * Connecting the editor to your redux store is maybe not what you really want.
* We could expose redux actions through the `Editor()` instance?

**Comment by inyono** [comment]
**TL;DR**. I advocate for keeping Redux including Provide but providing a way to access and modify the state without `connect`/`dispatch` so that developers don't need to use Redux. 

I like the way [react-native-router-flux](https://github.com/aksonov/react-native-router-flux) is doing things. They also use a Flux (don't know if it is Redux) architecture for the internal state management and expose `Actions` that contains the functions to dispatch actions. I.e., you can do something like this
```.js
import { Actions } from 'react-native-router'

Actions.push({}) // Dispatches the `PUSH` action
```
So to modify the router's state, you don't need to connect, let alone know anything about Flux/Redux.

Furthermore, they expose the action types so that you can [listen to it in your own Redux reducers](https://github.com/aksonov/react-native-router-flux/blob/master/docs/REDUX_FLUX.md) if you need to.

To give you access to the router's state, they pass props down to their children components. This, I guess, makes no sense for our case. But exporting `store.getState()` should be enough, shouldn't it?

**Comment by arekkas** [comment]
Cool, I think that is the best way forward, too! We allow the definition of custom middlewares, so you can do `getState()` but also have side-effects for actions (e.g. synch actions with backend stores) or for live edit. This plus `editor.dispatch.cellInsert({ data })` would be basically all we need.

**Comment by arekkas** [comment]
My current solution is:

```jsx
import Editor, { actions } from 'ory-editor'

const editor = new Editor()

editor.dispatch(actions.cell.insertAbove())
editor.dispatch(actions.cell.remove())
// ...
```

**Comment by inyono** [comment]
Yeah, I'm doing similar things in the creator. Maybe we should expose those directly, though? I.e. `editor.insertAbove()` instead of `editor.dispatch(actions.cell.insertAbove())`?

**Comment by arekkas** [comment]
We would still have to say what it is, right? So maybe: `editor.transform.cell.insertAbove()`? Not sure about the naming..

**Comment by inyono** [comment]
Ah yeah, sure. I just want to get rid of the dispatch / action creators :)

**Comment by arekkas** [comment]
Ok
